Update compose deployments based on feedback (#992)
owncloud/docs-ocis: modules/ROOT/partials/depl-examples/ubuntu-compose/shared-setup.adoc

* `SMTP_xxx` +
Define these settings according to your eMail configuration. With the settings defined, Infinite Scale is able to send notifications to users. If the settings are not defined, Infinite Scale will start, but notifications can't be sent.

NOTE: When not defining your own domain names, internal evaluation only domain names with self-signed certificates are used automatically.

== Certificate Generation Process

The recommended process to generate live certificates is the following:

. First, let LetsEncrypt generate "fake" certificates. These certifcates show that the process works but they cant be used in production. To do so, the environment variable `TRAEFIK_ACME_CASERVER` needs to be set to LetsEncrypt's https://letsencrypt.org/docs/staging-environment/[Staging Environment, window=_blank], see the `.env` file for the value to be set. By doing so, any restart after fixing issues will not count against the rate limit of LetsEncrypt.

> . Mandatory set the `TRAEFIK_ACME_MAIL` to a valid eMail address you own. When certificate issuing is triggered, LetsEncrypt checks in the request for creating valid certificates, if the response eMail address is valid and continues if so. If not, it logs an error and self-signed certificates are being used, see see xref:solving-first-startup-issues[Solving First Startup Issues].

Reviewer: ```suggestion
. The `TRAEFIK_ACME_MAIL` must be set to a valid eMail address you own. When certificate issuing is triggered, LetsEncrypt checks in the request for creating valid certificates, if the response eMail address is valid and continues if so. If not, it logs an error and self-signed certificates are being used, see see xref:solving-first-startup-issues[Solving First Startup Issues].
```

. xref:start-the-deployment[Start the Deployment].

. Then, if any issues occur, you need to xref:stop-the-deployment[Stop the Deployment] and fix them before continuing with the next step. See xref:solving-first-startup-issues[Solving First Startup Issues] for a list of common issues, 

. Finally, if there are no (more) issues which you can identify because `Fake LE intermediate X1` certificates have been generated (check the certificate issuer in the browser, Google for how to do so), you must xref:delete-cert-volume[delete the `cert-volume`] and set back the `TRAEFIK_ACME_CASERVER` environment variable to empty and startup the instance as described below.

== Start the Deployment

When you have finished the configuration, you can start the deployment by issuing the following command:

[source,bash]
----
docker compose up -d
----

This command will download all necessary containers and starts up the instance according your settings in the background (flag `-d`).

Check the logs::
--

> * See the xref:monitor-the-instance[Monitor the Instance] for more details on logging.

Reviewer: ```suggestion
* See xref:monitor-the-instance[Monitor the Instance] for more details on logging.
```

To get the state and the Container ID, issue one of the following commands:

.Long form, this command will print the required Container ID, among other data 
[source,bash]
----
docker ps -a
----

.Short form with only the Service name, State and Container ID, needs to be issued in `/opt/compose/ocis/{ocis_wopi}`:
[source,bash]
----
docker compose ps -a --format "table {{.Service}}\t{{.State}}\t{{.ID}}"
----

=== Logs

> Issue the following command to monitor a log which :

Reviewer: ```suggestion
Issue the following command to monitor a log:
```

I think this extra word can be removed.

* If there is no release gap, you can update by just stopping and starting the compose environment.
* For *any* https://owncloud.dev/ocis/release_roadmap/#dates[release gap], you must stop the compose environment with `docker compose down`, set the `OCIS_DOCKER_TAG` variable in the `.env` file accordingly, pull the new image with `docker compose pull` and start the compose environment with `docker compose up -d` again. For the last release, you can remove any setting of `OCIS_DOCKER_TAG` as `latest` is used by default.

== Certificate Renewal

> There is nothing special to do renewing certificates issued. Traefik takes care on the renewal process automatically. In case of a renewal issue, just restart the compose environment as traefik checks certificate expiry on startup.

Reviewer: ```suggestion
There is nothing special to do renewing certificates issued. Traefik takes care of the renewal process automatically. In case of a renewal issue, just restart the compose environment as Traefik checks certificate expiry on startup.
```
